Tests/Realm.Tests/Database/RealmDictionaryTests.cs

```diff
@@ -1073,6 +1073,235 @@ public void ObjectFromAnotherRealm_ThrowsRealmException()
             Assert.That(() => realm2.Write(() => obj2.EmbeddedObjectDictionary["foo"] = embeddedItem), Throws.TypeOf<RealmException>().And.Message.Contains("embedded object that is already managed"));
         }
 
+        public class DogDictionaryObject : RealmObject
+        {
+            public IDictionary<string, Dog> Dict { get; }
+        }
+
+        [Test]
+        public void DictionaryAsRealmQueryable_RaisesNotifications()
+        {
+            var obj = _realm.Write(() =>
+            {
+                return _realm.Add(new DogDictionaryObject());
+            });
+
+            var dogDict = obj.Dict;
+
+            var oldDogs = dogDict.AsRealmQueryable().Where(d => d.Age >= 5);
+
+            var changeSets = new List<ChangeSet>();
+            var token = oldDogs.SubscribeForNotifications((sender, changes, error) =>
+            {
+                if (changes != null)
+                {
+                    changeSets.Add(changes);
+                }
+            });
+
+            for (var i = 0; i < 10; i++)
+            {
+                _realm.Write(() => dogDict.Add("dog" + i, new Dog { Age = i }));
+                _realm.Refresh();
+
+                if (i >= 5)
+                {
+                    Assert.That(changeSets.Count, Is.EqualTo(i - 4));
+
+                    var changeSet = changeSets.Last();
+                    Assert.That(changeSet.InsertedIndices.Length, Is.EqualTo(1));
+                    Assert.That(changeSet.DeletedIndices, Is.Empty);
+                    Assert.That(changeSet.ModifiedIndices, Is.Empty);
+
+                    Assert.That(oldDogs.ElementAt(changeSet.InsertedIndices[0]).Age, Is.EqualTo(i));
+                }
+            }
+        }
+
+        [Test]
+        public void DictionaryAsRealmQueryable_WhenNotRealmDictionary_Throws()
+        {
+            var dict = new DictionariesObject().ObjectDictionary;
+
+            Assert.That(
+                () => dict.AsRealmQueryable(),
+                Throws.TypeOf<ArgumentException>().And.Message.Contains("dictionary must be an instance of RealmDictionary<IntPropertyObject>"));
```

Comment:
Yes, the "nameof" is getting the name of the variable that is used inside the function. The advantage here over using just a string is that if you do a rename this will be changed too